packages/editor/src/components/provider/disable-non-page-content-blocks.js

```diff
@@ -5,49 +5,68 @@ import { useSelect, useDispatch } from '@wordpress/data';
 import { store as blockEditorStore } from '@wordpress/block-editor';
 import { useEffect } from '@wordpress/element';
 
-const PAGE_CONTENT_BLOCKS = [
-	'core/post-title',
-	'core/post-featured-image',
+const CONTENT_ONLY_BLOCKS = [
 	'core/post-content',
+	'core/post-featured-image',
+	'core/post-title',
+	'core/template-part',
 ];
 
-function useDisableNonPageContentBlocks() {
-	const contentIds = useSelect( ( select ) => {
+/**
+ * Component that when rendered, makes it so that the site editor allows only
+ * page content to be edited.
+ */
+export default function DisableNonPageContentBlocks() {
```

Comment:
Not specific to this PR but I see this component as a hack. Also the `setBlockEditingMode` and `useBlockEditingMode` APIs have proven to be problematic in a lot of ways. Multiple conflicts and race conditions, I'd love to absorb these in the block editor framework/package instead (maybe through a setting or something) but we'd need to think properly about the API that we'd want to provide there.

Reply:
Yeah agreed and I keep running into limitations too. For example in this PR I want to disable insertion but not selection on the Template Part and disable all of its children. `contentOnly` doesn't really cut it for the former. There were similar issues with synced patterns.

We also have lots of ways to do similar things. I feel that with the right abstraction we could unify `tempateLock`, the `lock` attribute, and `setBlockEditingMode`.

Reply:
> We also have lots of ways to do similar things. I feel that with the right abstraction we could unify tempateLock, the lock attribute, and setBlockEditingMode.

I prefer templateLock instead because it's declarative while setBlockEditingMode is more subject to race conditions. I think the framework should be able to "guess" the mode, rather the consumer providing it for each block.